content/en/platform/corda/5.0/developing-applications/notaries/notary-plugin-cordapps.md

```diff
@@ -0,0 +1,69 @@
+---
+date: '2023-05-16'
+title: "Notary Plugin CorDapps"
+project: corda
+version: 'Corda 5.0'
+menu:
+  corda5:
+    identifier: notary-plugin-cordapps
+    parent: corda5-develop-notaries
+    weight: 4700
+section_menu: corda5
+---
+# Notary Plugin CorDapps 
+
+In Corda 5, notary functionality is provided in the form of plugin CorDapps. In theory, anyone can write a new notary protocol by implementing their own CorDapps. However, initially, it is expected to only have protocols provided by R3. For a given notary protocol, two CPBs (Corda Package Bundles) are expected to be required:
+
+* A **client**, or **application** CPB, which is used to generate a CPI (Corda Package Installer) associated with application virtual nodes. At a minimum, this contains a CPK (Corda Package) that has an initiating flow that is automatically invoked by the Corda 5 flow framework to initiate a notarization request.
+
+* A **notary server** CPB (Corda Package Bundle), which is used to generate a CPI associated with notary virtual nodes. At a minimum, this contains a CPK that has a responder flow to what is packaged in the client CPB.
+
+For Corda 5.0, only a single notary protocol is provided, the **non-validating notary protocol**.
+
+## Plugin Packaging
+The details of this architecture and the steps you need to take in order to successfully build your CorDapp will be outlined in the following sections. For information on how to deploy a functioning network, see the see the [Onboarding Notaries]({{< relref "../../application-networks/creating/notaries.md" >}}) section.
+
+{{< figure src="c5-non-validating-notary.png" figcaption="The CPKs, CPBs and CPIs involved in getting a functioning network that can run a notary (and by extension, UTXO ledger functionality)" alt="Corda 5 non-validating notary" >}}
+
+## Notary CPKs
+There are four modules/CPKs of relevance to the non-validating notary protocol, which are then packaged into two different CPBs. These are:
+
+* **`notary-common`**: This contains useful library functionality that is not specific to a particular notary protocol. For example, it defines the structure of an error that may be returned by a notary protocol. This is an example of something that must be standard cross-protocol, because the UTXO ledger depends on this format to perform processing, and the ledger has no visibility of the specific notary protocol that performs notarisation.
```

Comment:
```suggestion
* **`notary-common`** —  contains useful library functionality that is not specific to a particular notary protocol. For example, it defines the structure of an error that may be returned by a notary protocol. This is an example of something that must be standard cross-protocol, because the UTXO ledger depends on this format to perform processing, and the ledger has no visibility of the specific notary protocol that performs notarisation.
```